### PR TITLE
checking illegal characters in URI params, except for the URI param 'q'

### PR DESCRIPTION
**Pull request by kzangeli**
URI param names and values checked, except for the URI param 'q'

### CODE REVIEW

File: src/lib/rest/rest.cpp [hunk 1]
```diff
@@ -193,6 +193,21 @@ static int uriArgumentGet(void* cbDataP, MHD_ValueKind kind, const char* ckey, c
 
   LM_T(LmtUriParams, ("URI parameter:   %s: %s", key.c_str(), ciP->uriParam[key].c_str()));
 
+  //
+  // Now check the URI param has no invalid characters
+  // Except for the URI param 'q' that is not to be checked for invalid characters
+  //
+  if (key != "q")
```
Comment (fgalan) [inline src/lib/rest/rest.cpp]: Not only in `q` parameter, also in the case of `value` in scopes of `type` SCOPE_TYPE_SINGLE_FILTER (or something like that, I don't remember the `#define` used right now...).

File: src/lib/rest/rest.cpp [hunk 2]
```diff
@@ -193,6 +193,21 @@ static int uriArgumentGet(void* cbDataP, MHD_ValueKind kind, const char* ckey, c
...
+  {
+    if (forbiddenChars(ckey) || forbiddenChars(val))
```
Comment (fgalan) [inline src/lib/rest/rest.cpp]: This PR implement ilegal chracters checking in URI params in general. Thus, I think it deserves a line in CNR file (not tied to any specific issue).

Comment (kzangeli) [inline src/lib/rest/rest.cpp]: Yes, I was surprised to find out we didn't have any check of illegal chars in URI params ...

File: src/lib/rest/rest.cpp (hunk 1 above)
Comment (kzangeli) [inline src/lib/rest/rest.cpp]: This PR is only about the q parameter, as its title suggests.
Another PR about the scopes in the payload coming soon ...

NTC

File: src/lib/rest/rest.cpp (hunk 2 above)
Comment (kzangeli) [inline src/lib/rest/rest.cpp]: Issue #1141 created

Comment (kzangeli) [inline src/lib/rest/rest.cpp]: CNR fixed in 095fa6c

**Comment by fgalan** [comment]
LGTM

**Pull request merged**